v2/slogger/rolling_file_appender/rolling_file_appender.go

```diff
@@ -18,59 +18,72 @@
-// rotated.  Rotated log files will have suffix of the form
-// .YYYY-MM-DDTHH-MM-SS or .YYYY-MM-DDTHH-MM-SS-N (where N is an
-// incrementing serial number used to resolve conflicts) appended to
-// them.  Set maxFileSize to a non-positive number if you wish there
-// to be no limit.  maxRotatedLogs specifies the maximum number of
-// rotated logs allowed before old logs are deleted.  If
-// rotateIfExists is set to true and a log file with the same filename
-// already exists, then the current one will be rotated.  If
+// New creates a new RollingFileAppender.
+//
+// filename is path to the file to log to.  It can be a relative path
+// (with respect to the current working directory) or an absolute
+// path.
+//
+// maxFileSize is the approximate file size that will be allowed
+// before the log file is rotated.  Rotated log files will have suffix
+// of the form .YYYY-MM-DDTHH-MM-SS or .YYYY-MM-DDTHH-MM-SS-N (where N
+// is an incrementing serial number used to resolve conflicts)
+// appended to them.  Set maxFileSize to a non-positive number if you
+// wish there to be no limit.
+//
+// maxDuration is how long to wait before rotating the log file.  Set
+// to 0 if you do not want log rotation to be time-based.
+//
+// maxRotatedLogs specifies the maximum number of rotated logs allowed
+// before old logs are deleted.  Set to a non-positive number if you
+// do not want old log files to be deleted.
+//
+// If rotateIfExists is set to true and a log file with the same
+// filename already exists, then the current one will be rotated.  If
 // rotateIfExists is set to false and a log file with the same
 // filename already exists, then the current log file will be appended
 // to.  If a log file with the same filename does not exist, then a
 // new log file is created regardless of the value of rotateIfExists.
+//
 // As RotatingFileAppender might be wrapped by an AsyncAppender, an
 // errHandler can be provided that will be called when an error
-// occurs.  It can set to nil if you do not want to provide one.  The
-// return value headerGenerator, if not nil, is logged at the
+// occurs.  It can set to nil if you do not want to provide one.
+//
+// The return value headerGenerator, if not nil, is logged at the
 // beginning of every log file.
 //
 // Note that after creating a RollingFileAppender with New(), you will
 // probably want to defer a call to RollingFileAppender's Close() (or
 // at least Flush()).  This ensures that in case of program exit
 // (normal or panicking) that any pending logs are logged.
-func New(filename string, maxFileSize int64, maxRotatedLogs int, rotateIfExists bool, headerGenerator func() []string) (*RollingFileAppender, error) {
-	return NewWithStringWriter(filename, maxFileSize, maxRotatedLogs, rotateIfExists, headerGenerator, nil)
+func New(filename string, maxFileSize int64, maxDuration time.Duration, maxRotatedLogs int, rotateIfExists bool, headerGenerator func() []string) (*RollingFileAppender, error) {
```

Comment:
If a `maxFileSize` and a `maxDuration` are both supplied as positive values, will logs rotate when the first condition is met?

If a file is rotated due to hitting the `maxFileSize`, will the new file have it's `maxDuration` reset?

Can these behaviors be documented?